src/proxy/sessions/session.rs

```diff
@@ -100,36 +111,51 @@ impl Packet {
     pub fn contents(&self) -> &[u8] {
         &self.contents
     }
+
+    /// stop and record the stored [HistogramTimer]
+    pub fn stop_and_record(self) {
+        self.timer.stop_and_record();
+    }
 }
 
-impl Session {
-    /// new creates a new Session, and starts the process of receiving udp sockets
+pub struct SessionArgs {
+    pub metrics: Metrics,
+    pub proxy_metrics: ProxyMetrics,
+    pub filter_manager: SharedFilterManager,
+    pub from: EndpointAddress,
+    pub dest: Endpoint,
+    pub sender: mpsc::Sender<UpstreamPacket>,
+    pub ttl: Duration,
+}
+
+impl SessionArgs {
+    /// Creates a new Session, and starts the process of receiving udp sockets
     /// from its ephemeral port from endpoint(s)
-    pub async fn new(
-        base: &Logger,
-        metrics: Metrics,
-        filter_manager: SharedFilterManager,
-        from: EndpointAddress,
-        dest: Endpoint,
-        sender: mpsc::Sender<Packet>,
-        ttl: Duration,
-    ) -> Result<Self> {
+    pub async fn into_session(self, base: &Logger) -> Result<Session> {
```

Comment:
@XAMPPRocky this is definitely much cleaner 👍🏻 

Clippy complained about `new` as a function name, since it wasn't returning `Self`. I thought `into_session` made sense here (and I couldn't find any Rust naming conventions that said otherwise). PTAL, let me know what you think.

Reply:
Yeah, this works, I usually call it `run`, but there's no convention here.